### ISSUE TITLE
CS4: Mod injection and DLC are not entirely compatible.

### ISSUE THREAD

**Issue by AdmiralCurtiss** [AdmiralCurtiss/SenPatcher#129]
Reported by the creator of CS4 Restoration Project

> For some reason the DLC models (including Falcom's models, the PKG files included in assets/D3D11 and dlc/assets/D3D11) are not read. Probably the game reads the attachment list from t_attach.tbl files from data/text/dat_en and dlc/text/{dlc_number}/dat_en before the mod is loaded

Should be fixable, I just have to figure out what exactly is causing that.

**Comment by AdmiralCurtiss** [comment]
Ugh, I think I figured it out. It looks like when initializing the DLC files, the game opens the global `t_attach.tbl` and `t_item.tbl` files to append the DLC data onto it, and is modifying the 'open' file (really just a memory blob), and for some reason it fetches this already open handle via the same function that is used to open files the first time.

I did notice when writing the injection functions that there was some kind of cache in here that I didn't quite understand and it seemed to work fine to just ignore it, but it appears this is exactly what it's there for. So I probably just have to move my inject location to after the cache check instead of before it, and that should fix the bug here.
